examples/evm-to-substrate-react-example/src/evm-lib/state.ts

```diff
@@ -20,7 +21,7 @@ export type StateType = {
   apiState: string | null;
   selectedEvmConfig: EvmBridgeSetup | null;
   currentAccount: string | null;
-  signer: providers.JsonRpcSigner | null;
+  signer: NonceManager | null;
```

Comment:
@enemycnt I don't see why this change was introduced (using NonceManager instead of JsonRpcSigner)? Maybe I am missing something, can you just briefly describe me reasoning behind it.

Reply:
Sure, to make a deposit you need to send two transactions: `approve` and `deposit`. And in case of using JsonRpcSigner on deposit, I've got a nonce error. So it is a known problem. 
In `approveForBridge` I use `incrementTransactionCount()` function to increment nonce https://github.com/sygmaprotocol/sygma-sdk/blob/3e52c06160607f2c1faaf4068c7e360c030c0800/examples/evm-to-substrate-react-example/src/evm-lib/EvmContext.tsx#L316